AppCenterData/AppCenterData/MSData.m

```diff
@@ -556,6 +557,7 @@ - (void)upsertFromCosmosDbWithPartition:(NSString *)partition
       performCosmosDbOperationWithPartition:partition
                                  documentId:documentId
                                  httpMethod:kMSHttpMethodPost
+                                 httpClient:self.httpClientWithRetrier
```

Comment:
I might be reading this incorrectly, but this seems like an online operation (not a pending one). If so, then shouldn't this be the no retrier instance?

Reply:
good point, refactored the code, this method can be called from both CreateOrReplace and ProcessPendingOperations. therefore, it's required to path the desired httpClient as a parameter.